### PR TITLE
[new release] mirage, mirage-runtime, functoria and functoria-runtime (4.3.5)

### PR DESCRIPTION
**Pull request by hannesm**
The MirageOS library operating system

- Project page: <a href="https://github.com/mirage/mirage">https://github.com/mirage/mirage</a>
- Documentation: <a href="https://mirage.github.io/mirage/">https://mirage.github.io/mirage/</a>

#### Changed
- Remove OCaml dependency to allow OCaml 5 being chosen (mirage/mirage#1409 @palainp)
- Disallow empty key names (mirage/mirage#1407 @reynir)
- Allow tls-mirage 0.17 (mirage/mirage#1405 @hannesm)
- Allow tcpip 8.0 (mirage/mirage#1410 @hannesm)
- Fix typo in generated Makefile (mirage/mirage#1406 @hannesm)

### CODE REVIEW

**Comment by hannesm** [comment]
Please note that "test failures" are expected (due to opam 2.2 vs opam-monorepo), take a look into https://github.com/ocaml/opam-repository/pull/23504#issuecomment-1467860486 if interested.

**Comment by kit-ty-kate** [comment]
functoria-runtime seem to have a missing explicit dependency on ocamlfind:
```
#=== ERROR while compiling functoria-runtime.4.3.5 ============================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.5.0.0 | file:///home/opam/opam-repository
# path                 ~/.opam/5.0/.opam-switch/build/functoria-runtime.4.3.5
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p functoria-runtime -j 47
# exit-code            1
# env-file             ~/.opam/log/functoria-runtime-7-513c85.env
# output-file          ~/.opam/log/functoria-runtime-7-513c85.out
### output ###
# File "lib_runtime/functoria/dune", line 4, characters 12-17:
# 4 |  (libraries bytes cmdliner fmt))
#                 ^^^^^
# Error: Library "bytes" not found.
# -> required by library "functoria-runtime" in
#    _build/default/lib_runtime/functoria
# -> required by _build/default/META.functoria-runtime
# -> required by _build/install/default/lib/functoria-runtime/META
# -> required by _build/default/functoria-runtime.install
# -> required by alias install
```

File: packages/mirage/mirage.4.3.5/opam [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+dev-repo:     "git+https://github.com/mirage/mirage.git"
+license:      "ISC"
+tags:         ["org:mirage" "org:xapi-project"]
+doc:          "https://mirage.github.io/mirage/"
+available: opam-version >= "2.1.0"
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test & os != "macos"}
```
Comment (kit-ty-kate) [inline packages/mirage/mirage.4.3.5/opam:17]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test & os != "macos" & opam-version < "2.2"}
```

**Comment by hannesm** [comment]
> functoria-runtime seem to have a missing explicit dependency on ocamlfind:

I added such a dependency, and opened a PR upstream to remove the `bytes` dependency.

**Comment by hannesm** [comment]
IMHO this is fine to merge. Is there anything I can do to get it merged? Would the "& opam-version < 2.2" be required before a merge (if yes, let's add it)?

**Pull request merged**